sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/batch/BulkOperationRetryPolicy.java

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.batch;
+
+import com.azure.cosmos.BridgeInternal;
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.TransactionalBatchOperationResult;
+import com.azure.cosmos.implementation.HttpConstants.StatusCodes;
+import com.azure.cosmos.implementation.HttpConstants.SubStatusCodes;
+import com.azure.cosmos.implementation.IRetryPolicy;
+import com.azure.cosmos.implementation.ResourceThrottleRetryPolicy;
+import reactor.core.publisher.Mono;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * A container to keep retry policies and functions for bulk.
+ */
+final class BulkOperationRetryPolicy {
```

Comment:
Shouldn't this retry policy also inherit from RetryPolicyWithDiagnostics?

Reply:
Yeah, can help if/when we add some diagnostic info op wise.